lib/beaker/hypervisor/aws_sdk.rb

```diff
@@ -985,5 +986,16 @@ def load_fog_credentials(dot_fog = '.fog')
         :secret_key => default[:aws_secret_access_key],
       }
     end
+
+    # Adds port 8143 to host[:additional_ports]
+    # if master, database and dashboard are not on same instance
+    def test_split_install
+      @hosts.each do |host|
+        mono_roles = ['master', 'database', 'dashboard']
+        if (host[:roles] & mono_roles).size != 3 and (host[:roles] & mono_roles).any?
```

Comment:
@doug-rosser the bitwise `&` of `host[:roles]` with `mono_roles` will bring down the size of the array to 3 if master, dashboard and database is provided on the same host. And if that's the case, it's size will also be 3 which would fail the first part of the if statement and therefore `8143` will not be open on that host.